src/DependencyInjection/Compiler/ContaoRootDirPass.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/*
+ * This file is part of Contao.
+ *
+ * Copyright (c) 2005-2016 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao\CoreBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Canonicalizes the contao.root_dir path.
+ *
+ * @author Leo Feyer <https://github.com/leofeyer>
+ */
+class ContaoRootDirPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $container->setParameter('contao.root_dir', realpath($container->getParameter('contao.root_dir')));
```

Comment:
Is `realpath` the right function to use here? It does not only resolve `/../` but also expands symlinks, checks for existence of the directory, may normalize the character case and could change unix style paths to windows style.